### PR TITLE
docs(tokens) Updates migration instructions.

### LINKED ISSUE #4218: Document how to upgrade React tokens
**Is this a new page or a change to an existing one?**
We need to give some guidance on how to handle upgrading React tokens. They should be pointing to the var rather than the value so that dark theme will work properly.

For things like breakpoints, spacers, and widths we may be able to make a reliable substitution with a code mod.

For colors, let's add a code mod pointing to a new "placeholder" token set to a hot pink highlight color. This will make it easy to find places where semantic color tokens need to be selected.

Here's a start to the information that could be on this page. https://docs.google.com/document/d/1kjqv-gyuf2k9fewlUpnV3kpCmFbM_q6YhutPsjlOu88/edit?usp=sharing

### PR DESCRIPTION
**Pull request by edonehoo**
Closes #4218, adds to existing tokens docs

### CODE REVIEW

**Comment by edonehoo** [comment]
@srambach RE https://github.com/patternfly/patternfly-org/issues/4218, were you thinking of a different direction than this pr goes? Much of the info from the linked doc in the issue was already on the site, but do you think it would make more sense in a different page or with more detail? Added some more details and specifics, though!

I saw the codemods notes in the original doc, but they seemed like observations more than things we want to communicate on org

**Review by dlabaj — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Comment by adamviktora** [comment]
Following on what Sarah has written about React tokens - we just discussed that on a working session, and it would be great to add some notes like this:

- React token names are based on the CSS variable names, but instead of `--` symbol, `_` is used plus the starting `--pf-` is removed. So for example `--pf-t--global--spacer--sm` becomes a `t_global_spacer_sm` react token.
- React token can be imported from its file directly: `import t_global_spacer_sm from '@patternfly/react-tokens/dist/esm/t_global_spacer_sm'` or from the whole package: `import { t_global_spacer_sm } from '@patternfly/react-tokens'`
- React token is an object having name, value and var properties:
```
const t_global_spacer_sm = {
  "name": "--pf-t--global--spacer--sm",
  "value": "0.5rem",
  "var": "var(--pf-t--global--spacer--sm)"
}
```

**Comment by kmcfaul** [comment]
@adamviktora Is this a blocker or could it be done in a follow up?

**Comment by adamviktora** [comment]
@kmcfaul It can be a follow up for sure

**Comment by srambach** [comment]
@kmcfaul I think @adamviktora 's suggestions are the core of this issue so I think it's worth doing before merging this rather than making another issue.

**Comment by edonehoo** [comment]
@adamviktora ty for this context! 
@srambach I integrated Adam's comment, but I can't tell if it's enough extra context on its own to help guide people that will actually use this info. WDYT? 

If we do want to go further in these docs, or if I'm misunderstanding any of these details, it would probably be good to arrange time for a call to help me understand this more deeply (also cc @nicolethoen) !

**Review by adamviktora — COMMENTED** [review]
I really like the updates, I think it explains React tokens pretty well. Two comments bellow:

File: packages/documentation-site/patternfly-docs/content/get-started/upgrade.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ PatternFly 6 supports our new design token system, which changes variable names
 
 Wherever you have any custom CSS overrides that reference PatternFly class names or CSS variables, you should carefully review them and make updates to ensure that they align with our token variables, which are outlined in our [tokens documentation](/tokens/all-patternfly-tokens). As much as possible, we recommend removing your CSS overrides so that your product upgrade experience will be smoother for future releases.
 
-If your product uses a custom solution to replicate PatternFly styling (without using PatternFly components), then it will need to be reskinned. We recognize that this may be a large undertaking, so we encourage you to reach out to the PatternFly team so that we support this work.
+If your product uses a custom solution to replicate PatternFly styling (without using PatternFly components), then it will need to be re-skinned. We recognize that this may be a large undertaking, so we encourage you to reach out to the PatternFly team so that we can help support this work.
```
[the remark this answers is not in the document]
Comment (edonehoo) [reply packages/documentation-site/patternfly-docs/content/get-started/upgrade.md:61]: thanks for all of this info!! Is that this codemod? https://github.com/patternfly/pf-codemods?tab=readme-ov-file#tokens-update. I linked to that one in my last commit, but can adjust if needed

File: packages/documentation-site/patternfly-docs/content/get-started/upgrade.md [hunk 2]
```diff
@@ -71,4 +71,28 @@ When using this codemod, keep the following guidance in mind:
 - Add the `--fix` flag to allow the codemod to fix issues where possible.
 - This utility performs a simple ‘find and replace’, so it's possible that it will inadvertently identify code that is formatted similarly to a PatternFly class name, but is not one. You should check to ensure that this doesn't cause any unintentional changes.
 
-For more details, you can refer to [the README file for this codemod](https://github.com/patternfly/pf-codemods/tree/main/packages/class-name-updater).
\ No newline at end of file
+For more details, you can refer to [the README file for this codemod](https://github.com/patternfly/pf-codemods/tree/main/packages/class-name-updater).
+
+### Update React tokens
```
[the remark this answers is not in the document]
Comment (edonehoo) [reply packages/documentation-site/patternfly-docs/content/get-started/upgrade.md]: Ah yes, I see what you mean, good point. I moved it!

File: packages/documentation-site/patternfly-docs/content/get-started/upgrade.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamviktora) [reply packages/documentation-site/patternfly-docs/content/get-started/upgrade.md:61]: Yes, that's the codemod. The class-name-updater is not limited to just css. Thanks for doing all the updates!

**Review by kmcfaul — APPROVED** [review]
🎉

**Review by srambach — APPROVED** [review]
I think this looks good, thank you! 🩷

**Pull request merged**